lib.nix

```diff
@@ -174,7 +174,7 @@ rec
             attrs =
               # Since we pretend everything is a lib, we remove any mentions
               # of binaries
-              removeAttrs cargotoml [ "bin" "example" "lib" "test" "bench" "default-run" ]
+              removeAttrs cargotoml [ "bin" "example" "test" "bench" "default-run" ]
```

Comment:
what happens if the project mentions a `[lib]` that isn't `src/lib.rs`? Won't cargo try to build that and, not finding any files, fail?

Reply:
To be honest, this change in particular is the one I'm not sure about. In order for WASM dependencies to be built, `lib.crate-type` needs to be retained. This was merely the quickest fix I could think of, I didn't really consider the other effects, and perhaps there's a better way to achieve the same goal.

Reply:
Any example which uses `wasm-bindgen` for instance:
https://github.com/rustwasm/wasm-bindgen/tree/main/examples